### ISSUE TITLE
LEDs - 3 generations, 3 different issues.

### ISSUE THREAD

**Issue by Wh1terat** [jberclaz/modelh#7]
I've got 3x Model Ms I've fitted with the Model-H board. 
All working nicely as a keyboard but all have different issues relating to the Num/Caps/Scroll lock LEDs.

Any suggestions?

### 1988 "Silver Square" 
This previously had the BLUSB controller from [mind_prepared](https://deskthority.net/memberlist.php?mode=viewprofile&u=6240&sid=bf6b078a8baf9f7f865929683246edd5) but unfortunately the LIPO did what LIPOs like to do and expanded. It has the replacement "lower power" LED strip from blusb.

![m-1](https://github.com/user-attachments/assets/ba0f4e65-42f1-4039-aa1e-37b29028e3d9)
![m-1-board](https://github.com/user-attachments/assets/c6050ede-a88e-4075-9472-34234725a690)
![IMG_0793 Medium](https://github.com/user-attachments/assets/296833cd-294f-4576-84ed-2ffec7edce1f)

Num Lock: No LED
Caps Lock: Turns on Num Lock LED
Scroll Lock: No LED

### 1986 "Silver Square"  (Very early - with "foam wedge").

![m-2](https://github.com/user-attachments/assets/cfe56ca7-2843-4e25-aab4-924a903f5e8c)
![m-2-board](https://github.com/user-attachments/assets/fe2d8c0e-e795-4f5d-9314-e98412c1e4fd)
![IMG_0791 Medium](https://github.com/user-attachments/assets/e73140ee-af43-4ea3-bcc4-24798a8b3841)

Num Lock: No LED
Caps Lock: No LED
Scroll Lock: No LED

### 1990 "Grey Oval"

![m-3](https://github.com/user-attachments/assets/deaa9235-abd9-4f11-92e1-d1222c74772b)
![m-3-board](https://github.com/user-attachments/assets/11acbb04-c2f5-46f8-8fe1-584e303343c7)
(Sorry no pic of LED strip - but uses FPC rather than JST connector, like example in readme)

Num Lock: No LED
Caps Lock: Works
Scroll Lock: No LED

**Comment by jberclaz** [comment]
Good to hear that you were able to use the Model H in three different generations of Model M. For your issues with the LEDs, I would suggest the following:
1. check the cable with a multimeter. I had one keyboard where the ribbon cable to the LEDs was broken (even though there was no obvious visible damage).
2. with a multimeter again, check the current at the LED connectors on the board. You should get current into a different one of the 3 outputs for each LED (num, caps and scroll). If you don't have that, I assume the board is broken.
3. you can switch the lights in the firmware. For that you'll have to recompile QMK. Open the file `keyboards/ibm/model_m/modelh/keyboard.json` and switch the values found in the `indicators` block. You'll find instructions on how to compile the firmware on the repo.
Hope this helps.

**Comment by Wh1terat** [comment]
Ok so keyboard 1 and 2 are the same problem. The PCB has the wrong pinout 🤦‍♂️ 
Re-pinned the JST cables to get it working:
![ibm-re-pin](https://github.com/user-attachments/assets/ea835d96-8cd4-47dd-bb4d-17a1baf2bb4c)

Keyboard 3 was just a dirty FPC connector 🙈 

I really don't like the way the FPC cable has to twist, if the board were ever to be revised it would be nicer to use a full size board footprint like original PCB and move the LED FPC socket to the top and orientated properly AND leave a footprint on the right for the JST connector for these earlier boards (with correct pinout).

**Comment by jberclaz** [comment]
Glad to hear you made it work.

I agree that the position of the LED connector is not great. Though, I believe the design is intended to match the shape and size of some of the later boards, which were smaller.
![boards](https://github.com/user-attachments/assets/d4e4ee3d-d556-4326-9ad3-a55fa04629bf)

**Comment by Wh1terat** [comment]
I do wonder which generation board that top one is, certainly a less common one I would say.
Especially when looking at photos on: https://sharktastica.co.uk/wiki?id=modelmenhanced

I recently came across another project focussing more on wireless keyboard (and thus ZMK based) which looks to incorporate a number of improvements:

https://github.com/NilsFC/Model-X

**Comment by SharktasticA** [comment]
> I do wonder which generation board that top one is, certainly a less common one I would say. Especially when looking at photos on: https://sharktastica.co.uk/wiki?id=modelmenhanced

I've seen keyboards with that controller card (typically P/N 1395601) from keyboards made between 1990 and 1991 which would be within the "second gen" (this card style is currently absent from my website because I don't have any good, usable photos yet). Here's a dated example from the [Ardent Tool](https://www.ardent-tool.com/keyboard/Model_M.html):

![image](https://github.com/user-attachments/assets/4b75f946-cc4b-416d-b1c6-777cf970d26c)

**Comment by Wh1terat** [comment]
Thanks @SharktasticA
Still amazes me how many revisions of the controller exist. Talk about reinventing the wheel 😄 

If you'd like any higher-res photos of the boards/plates/etc I have (the black/purple PCB perhaps) do let me know.

**Comment by jberclaz** [comment]
> Still amazes me how many revisions of the controller exist.

True, out of the 6 Ms I have, none of them has the same controller.
